### DISCUSSION TITLE
Same cookie being set for two apps running in different ports

### DISCUSSION THREAD

**Discussion by kombos** [vercel/next.js#64097] [App Router]
I'm building a NextJS 14.1.3 app, where in one of the server actions, cookies are set in the client: 

```
export async function otpAction(username, otp) {
  let playResponse;
  const cookieStore = cookies();

  try {
    playResponse = await postValidateOtp(username, otp);
    // take 'Set-Cookie' header value
    const playResponseCookies = playResponse.headers.get("set-cookie");
    // set set-cookie header string to response cookie
    if (playResponseCookies) {
      let cookieObject = parseHeaderToCookie(playResponseCookies);
      cookieStore.set(cookieObject);
    }

    if (!playResponse.ok) {
      let playResponseBody = await playResponse.json();
      if (
        ["VALIDATE_OTP_FAILED", "INVALID_OTP"].some((str) =>
          playResponseBody?.message?.includes?.(str)
        )
      )
        return { ok: false, status: 401 };
      return { ok: false, status: playResponse?.status };
    }
  } catch (error) {
    return { ok: false, status: 500 };
  }

  // dont return anything since can't return non-serializable object like fetch response
  return { ok: true, status: playResponse?.status || 500 };
}
```
the attributes that are usually set for this cookie are: 
SameSite=lax
HTTPOnly=true
path=/
secure=true

i'm running this nextjs app in dev mode (next dev) in localhost:3000, but on setting this cookie, it is also setting the same cookie in localhost:3001 where a different app is running. both these apps are opened in the same chrome browser as different tabs. is this normal behavior? if not how to fix this?

**Comment by icyJoseph** [comment]
Cookies do not provide by port isolation:

- https://stackoverflow.com/a/16328399
- https://levelup.gitconnected.com/be-careful-running-multiple-localhost-apps-8e69e0a0293b
